client/db/src/lib.rs

```diff
@@ -1257,13 +1261,23 @@ impl<Block: BlockT> Backend<Block> {
 
 			let finalized = if operation.commit_state {
 				let mut changeset: sc_state_db::ChangeSet<Vec<u8>> = sc_state_db::ChangeSet::default();
+				let mut ops: u64 = 0;
+				let mut bytes: u64 = 0;
 				for (key, (val, rc)) in operation.db_updates.drain() {
 					if rc > 0 {
+						ops += 1;
+						bytes += key.len() as u64 + val.len() as u64;
+
 						changeset.inserted.push((key, val.to_vec()));
 					} else if rc < 0 {
+						ops += 1;
+						bytes += key.len() as u64;
+
 						changeset.deleted.push(key);
 					}
 				}
+				self.state_usage.tally_writes(ops, bytes);
```

Comment:
Is this valid only for `PrunningMode::ArchiveAll`? Maybe you should make it explicit in that case.

Reply:
I don't think so, it is valid for any prunning mode